doc/source/cluster/kubernetes/k8s-ecosystem/yunikorn.md

```diff
@@ -0,0 +1,158 @@
+
+
+```shell
+# Path: kuberay/ray-operator/config/samples
+# Includes the necessary labels for the Apache YuniKorn scheduler gang scheduling:
+# - `ray.io/gang-scheduling-enabled`
+# - `yunikorn.apache.org/app-id`
+# - `yunikorn.apache.org/queue`
+wget https://raw.githubusercontent.com/ray-project/kuberay/v1.2.2/ray-operator/config/samples/ray-cluster.yunikorn-scheduler.yaml
+kubectl apply -f ray-cluster.yunikorn-scheduler.yaml
+```
+
+Because the queue has a capacity of 4 CPU and 6Gi of RAM, this resource should schedule successfully without any issues.
+
+```shell
+$ kubectl get pods
+
+NAME                                  READY   STATUS    RESTARTS   AGE
+test-yunikorn-0-head-98fmp            1/1     Running   0          67s
+test-yunikorn-0-worker-worker-42tgg   1/1     Running   0          67s
+test-yunikorn-0-worker-worker-467mn   1/1     Running   0          67s
+```
+
+Next, add an additional RayCluster with the same configuration of head and worker nodes, but with a different name:
+
+```shell
+# Path: kuberay/ray-operator/config/samples
+# Includes the `ray.io/scheduler-name: volcano` and `volcano.sh/queue-name: kuberay-test-queue` labels in the metadata.labels
+# Replaces the name to test-yunikorn-1
+sed 's/test-yunikorn-0/test-yunikorn-1/' ray-cluster.yunikorn-scheduler.yaml | kubectl apply -f-
+```
+
+Now all the pods for `test-yunikorn-1` are in the `Pending` state:
+
+```shell
+$ kubectl get pods
+
+NAME                                      READY   STATUS    RESTARTS   AGE
+test-yunikorn-0-head-98fmp                1/1     Running   0          4m22s
+test-yunikorn-0-worker-worker-42tgg       1/1     Running   0          4m22s
+test-yunikorn-0-worker-worker-467mn       1/1     Running   0          4m22s
+test-yunikorn-1-head-xl2r5                0/1     Pending   0          71s
+test-yunikorn-1-worker-worker-l6ttz       0/1     Pending   0          71s
+test-yunikorn-1-worker-worker-vjsts       0/1     Pending   0          71s
+tg-test-yunikorn-1-headgroup-vgzvoot0dh   0/1     Pending   0          69s
+tg-test-yunikorn-1-worker-eyti2bn2jv      1/1     Running   0          69s
+tg-test-yunikorn-1-worker-k8it0x6s73      0/1     Pending   0          69s
+```
+
+Because the new cluster requires more CPU and RAM than our queue allows, even though one of the pods would fit in the remaining 1 CPU and 2Gi of RAM, none of the cluster's pods are placed until there is enough room for all the pods. Without using Volcano for gang scheduling in this way, one of the pods would ordinarily be placed, leading to the cluster being partially allocated, and some jobs (like [Horovod](https://github.com/horovod/horovod) training) being stuck waiting for resources to become available.
+
+Delete the first RayCluster to make space in the queue:
+
+```shell
+kubectl delete raycluster test-yunikorn-0
+```
+
+Now all the pods for the second cluster changed to the `Running` state, because enough resources are now available to schedule the entire set of pods:
```

Comment:
nit
```suggestion
Now all the Pods for the second cluster changed to the `Running` state, because enough resources are now available to schedule the entire set of Pods:
```